types/api.d.ts

```diff
@@ -0,0 +1,49 @@
+import {Uri} from 'vscode';
+
+declare interface BasToolkit {
+    getExtensionAPI: <T>(extensionId: string) => Promise<T>;
+    getAction: (actionId: string) => IAction | undefined;
+    getParameter: (parameterName: string) => Promise<string | undefined>;
+
+    actions: {
+        performAction: (action: IAction, options?: any) => Thenable<void>;
+        ExecuteAction: { new(): IExecuteAction };
```

Comment:
add @deprecated annotation in comment on these classes as we intend to remove them.
Maybe even some deprecation related comment text information.